src/middlewared/middlewared/plugins/failover_/event.py

```diff
@@ -659,6 +659,11 @@ def vrrp_master(self, job, fobj, ifname, event):
 
         self.run_call('interface.persist_link_addresses')
 
+        try:
+            self.run_call('failover.check_reboot_required')
```

Comment:
I know this check _SHOULD_ be quick, we really need to review all it calls to make sure we don't cause any unexpected delays in the failover process. Also, please add a logging statement here so we can see it in the logs. (i.e. `logger.info('Checking if reboot is required')`)

Reply:
I will do that and i checked to make sure we have timeouts for remote calls in case they get stuck up for some reason and verified those calls as well when remote was down to make sure we are not stuck